### PR TITLE
In mysqlxx::Pool: fix for reconnection problem

### PR DESCRIPTION
**Pull request by Akazz**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed problem when ClickHouse fails to resume connection to MySQL servers.

Detailed description / Documentation draft:
Under certain conditions CH may become unable to resume its connections to mysql when mysql server is used as a data source (external dictionaries, tables, etc.)

### CODE REVIEW

File: base/mysqlxx/Pool.cpp [hunk 1]
```diff
@@ -150,14 +151,22 @@ Pool::Entry Pool::tryGet()
...
+        if ((*connection_it)->ref_count == 0)
         {
-            Entry res(connection, this);
-            return res.tryForceConnected() ? res : Entry();
+            Entry res(*connection_it, this);
+            if (res.tryForceConnected())  /// Will try to reestablish connection as well
+                return res;
+
+            /// This one is disconnected for good and so must be removed
+            ::delete *connection_it;  /// TODO: Manual memory management is awkward (matches allocConnection() method)
```
Comment (alexey-milovidov) [inline base/mysqlxx/Pool.cpp:165]: So, the fix is in these two lines?

Comment (Akazz) [reply base/mysqlxx/Pool.cpp:165]: Basically yes, it is important to get rid of such unrecoverable connections.

**Comment by Akazz** [comment]
Failing [**Functional stateless tests**](https://clickhouse-test-reports.s3.yandex.net/17681/62649d0dd0777f3080319830601e4abc56f4d363/functional_stateless_tests_(memory).html#fail1) in 62649d0 seem to be flapping in *01304_direct_io*.

Generally, this PR looks safe to be merged in.

**Comment by Akazz** [comment]
Tests are OK

**Pull request merged**